tests/rptest/services/chaos/workloads/service_base.py

```diff
@@ -86,7 +86,16 @@ def _request(self, verb, node, path, timeout_sec=10, **kwargs):
         kwargs["timeout"] = timeout_sec
         url = self._remote_url(node, path)
         self.logger.debug(f"dispatching {verb} {url}")
-        r = requests.request(verb, url, **kwargs)
+        existing_exc = sys.exc_info()[1]
+        try:
+            r = requests.request(verb, url, **kwargs)
+        except requests.RequestException as e:
+            # Request exceptions (e.g. timeouts) come with the context attribute
+            # that is already set to some internal requests exception. This is not
+            # nice because the context corresponding e.g. to a test error will be
+            # lost. Set it explicitly here.
```

Comment:
Interesting, looks like this applies only to request errors, not connection errors. Try with `requests.request('GET', 'http://127.0.0.1:8080/', timeout=10)` and `nc -l 8080` in another terminal - the original context is lost.

Reply:
I tried this and it still works as appropriate:
```
HTTPConnectionPool(host='127.0.0.1', port=8080): Read timed out. (read timeout=10)
-- the exception above happened when handling the exception below --
HTTPConnectionPool(host='127.0.0.1', port=8080): Read timed out. (read timeout=10)
-- the exception above happened when handling the exception below --
timed out
-- the exception above happened when handling the exception below --
HTTPConnection.getresponse() got an unexpected keyword argument 'buffering'
-- the exception above happened when handling the exception below --
division by zero
```
I tried this both on my machine and in a ducktape run to see if python requests module version matters. It is 2.32.2 on the host and 2.24.0 in ducktape environment. We don't seem to specify exact version in `tests/setup.py` however, so you might be unlucky to get a buggy one as a dependency. In bazel builds it looks fixed in `MODULE.bazel.lock`.

Anyway, in the environment where you can reproduce the problem, what is `requests.__version__`?

Reply:
Ok, I found it. The culprit is this (rather ugly) piece of code: https://github.com/urllib3/urllib3/blob/1.26.5/src/urllib3/connectionpool.py#L433-L445

If `conn.getresponse()` throws an exception, stray TypeError due to an unexpected keyword argument gets included into the context chain. To avoid that, they re-raise the actual exception with `raise e from None`, which sets `__suppress_context__` to True. The `__context__` field is preserved (which is what you are seeing), but traceback-printing routines stop at that exception, thereby hiding the original cause.